flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -766,13 +776,12 @@ private void returnExcessMemorySegments() {
 
     @GuardedBy("availableMemorySegments")
     private boolean hasExcessBuffers() {
-        return numberOfRequestedOverdraftMemorySegments > 0
-                || numberOfRequestedMemorySegments > currentPoolSize;
+        return numberOfRequestedOverdraftMemorySegments > 0;
```

Comment:
IIUC, If we allow all buffers that exceed poolSize to be `overdraft`, then `numberOfRequestedMemorySegments > currentPoolSize` will no happen, right? @1996fanrui can you help me to double confirm this, thx~

Reply:
I have checked, and it's ok from my side. However, I prefer inviting more experts to review this PR, it will be more reliable.

Hi @pnowojski @akalash , would you mind take a look this PR in your free time? thanks.

Reply:
> I prefer inviting more experts to review this PR, it will be more reliable.

Sure, It will be better~

Reply:
Technically, this change is correct but since `numberOfRequestedOverdraftMemorySegments` and `numberOfRequestedMemorySegments` can be changed independently I would prefer to leave it as is. And we can make this change as soon as we get rid of `numberOfRequestedOverdraftMemorySegments`(if we will do it at all).
Maybe I am a little paranoic about this but the current contract between `numberOfRequestedOverdraftMemorySegments` and `numberOfRequestedMemorySegments` looks a little bit fragile so I am afraid that it is easy to introduce the bug when `numberOfRequestedOverdraftMemorySegments > 0 == false` while `umberOfRequestedMemorySegments > currentPoolSize == true`

Reply:
> we can make this change as soon as we get rid of numberOfRequestedOverdraftMemorySegments(if we will do it at all).

Yes, I'm sure we'll be removing `numberOfRequestedOverdraftMemorySegments` soon as keeping it would complicate the contract a bit.

I agree that it's easy to introduce some hard-to-find bugs, especially since we'll be refactoring it soon. I will revert the change here.